extension/observer/ecsobserver/target.go

```diff
@@ -127,3 +140,51 @@ var (
 func sanitizeLabelName(s string) string {
 	return invalidLabelCharRE.ReplaceAllString(s, "_")
 }
+
+type fileSDTarget struct {
+	Targets []string          `yaml:"targets" json:"targets"`
+	Labels  map[string]string `yaml:"labels" json:"labels"`
+}
+
+func targetsToFileSDTargets(targets []PrometheusECSTarget, jobLabelName string) ([]fileSDTarget, error) {
+	var converted []fileSDTarget
+	omitEmpty := []string{labelJob, labelServiceName}
+	for _, t := range targets {
+		labels := t.ToLabels()
+		address, ok := labels[labelAddress]
+		if !ok {
+			return nil, fmt.Errorf("address label not found for %v", labels)
+		}
+		delete(labels, labelAddress)
+		// Remove some labels if their value is empty
+		for _, k := range omitEmpty {
+			if v, ok := labels[k]; ok && v == "" {
+				delete(labels, k)
+			}
+		}
+		// Rename job label as a workaround for https://github.com/open-telemetry/opentelemetry-collector/issues/575
```

Comment:
Can you provide more detail on what this is doing and why?  It looks like the referenced issue has been resolved.

Reply:
Actually that issue is two issues, one is dropping job label, another is providing job label in sd or relabel it. I think the comment in that issue explains it pretty well https://github.com/open-telemetry/opentelemetry-collector/issues/575#issuecomment-814558584